packages/components/src/input-control/types.ts

```diff
@@ -62,10 +60,10 @@ interface BaseProps {
 	size?: Size;
 }
 
-export type InputChangeCallback<
-	E = ChangeEvent< HTMLInputElement > | PointerEvent< HTMLInputElement >,
-	P = {}
-> = ( nextValue: string | undefined, extra: { event: E } & P ) => void;
+export type InputChangeCallback< P = {} > = (
+	nextValue: string | undefined,
+	extra: { event: SyntheticEvent } & P
+) => void;
```

Comment:
I changed `InputControl`'s `onChange` callback type to be less opinionated. 

Typing the event attribute as `PointerEvent<T> | ChangeEvent<T>` isn't great for maintainability as it means that if a new way to input a value is added to `InputControl`, `NumberControl` or `UnitControl` then there will be a BC-breaking type change.

Consumers of the component don't need to know the specifics of the event beyond that it's a synthetic event (not a native event). Consumers can use `is` to determine the event type if necessary.